src/HDF5.jl

```diff
@@ -1258,6 +1258,13 @@ function read{A<:FixedArray}(obj::DatasetOrAttribute, ::Type{Array{A}})
     ret
 end
 
+# Read a scalar reference objected from an attribute
+function read{T<:HDF5ReferenceObj}(obj::HDF5Attribute, ::Type{T})
+    refs = Array{T}(1)
```

Comment:
True, but more lifting would be required since `getindex` doesn't support Ref at the moment.